pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/ASTTypeWhenBlock.java

```diff
@@ -4,29 +4,20 @@
 
 package net.sourceforge.pmd.lang.apex.ast;
 
-import com.google.summit.ast.Node;
-import org.apache.commons.lang3.reflect.FieldUtils;
+import com.google.summit.ast.statement.SwitchStatement;
 
-public final class ASTTypeWhenBlock extends AbstractApexNode.Single<Node> {
+public final class ASTTypeWhenBlock extends AbstractApexNode.Single<SwitchStatement.WhenType> {
 
-
-    ASTTypeWhenBlock(Node node) {
-        super(node);
+    ASTTypeWhenBlock(SwitchStatement.WhenType whenType) {
+        super(whenType);
     }
 
     public String getType() {
-        // return String.valueOf(node.getTypeRef());
-        // TODO(b/239648780)
-        return null;
+        return node.getType().asCodeString();
```

Comment:
nit: another case where I'm wondering if we want to use the printing-as-code methods

Reply:
Discussed externally; resolved for now